src/entities/Snapshot/utils.ts

```diff
@@ -87,3 +92,121 @@ export function groupProposalsByMonth(proposals: Partial<SnapshotProposal>[], fi
+  const values = await Promise.all([
+    await SnapshotGraphql.get().getStatus(),
+    await SnapshotGraphql.get().getSpace(spaceName),
+  ])
+  const snapshotStatus = values[0]
+  const snapshotSpace = values[1]
+  if (!snapshotSpace) {
+    throw new Error(`Couldn't find snapshot space ${spaceName}. 
+      \nSnapshot response: ${JSON.stringify(snapshotSpace)}
+      \nSnapshot status: ${JSON.stringify(snapshotStatus)}`)
+  }
+  return { snapshotStatus, snapshotSpace }
+}
+
+export function filterDelegationTo(delegations: Delegation[], space: string): Delegation[] {
+  if (delegations.length > 1) {
+    return delegations.filter((delegation) => delegation.space === space)
+  }
+
+  return delegations
+}
+
+export function filterDelegationFrom(delegations: Delegation[], space: string): Delegation[] {
+  if (delegations.length === 0) {
+    return []
+  }
+
+  const unique_delegations = new Map<string, Delegation>()
+
+  for (const deleg of delegations) {
+    if (unique_delegations.has(deleg.delegator)) {
+      if (unique_delegations.get(deleg.delegator)?.space !== space) {
+        unique_delegations.set(deleg.delegator, deleg)
+      }
+    } else {
+      unique_delegations.set(deleg.delegator, deleg)
+    }
+  }
+
+  return Array.from(unique_delegations.values())
+}
+
+async function getDelegations(query: string, variables: any): Promise<DelegationResult> {
+  try {
+    const delegates = await SnapshotGraphql.get().fetchDelegates(query, variables)
+    if (!delegates) {
+      return EMPTY_DELEGATION
+    }
+    const filteredDelegatedFrom = filterDelegationFrom(delegates.delegatedFrom, SNAPSHOT_SPACE)
+    return {
+      delegatedTo: filterDelegationTo(delegates.delegatedTo, SNAPSHOT_SPACE),
+      delegatedFrom: filteredDelegatedFrom.slice(0, 99),
+      hasMoreDelegatedFrom: filteredDelegatedFrom.length > 99,
+    }
+  } catch (error) {
+    console.error(error)
+    return EMPTY_DELEGATION
+  }
+}
+
+const DELEGATIONS_ON_PROPOSAL_QUERY = `
+query ($space: String!, $address: String!, $blockNumber: Int) {
+  delegatedTo: delegations(block:{number: $blockNumber},where: { space_in: ["", $space], delegator: $address }, orderBy: timestamp, orderDirection: desc) {
+    delegator
+    delegate
+    space
+    timestamp
+  },
+  delegatedFrom: delegations(block:{number: $blockNumber},where: { space_in: ["", $space], delegate: $address }, orderBy: timestamp, orderDirection: desc) {
+    delegator
+    delegate
+    space
+    timestamp
+  }
+}`
+
+export async function getDelegationsOnProposal(address?: string | null, proposal?: ProposalAttributes | null) {
+  if (!SNAPSHOT_SPACE || !address || !proposal) {
+    return EMPTY_DELEGATION
+  }
+  const variables = {
+    address: address.toLowerCase(),
+    space: SNAPSHOT_SPACE,
+    blockNumber: proposal.snapshot_proposal.snapshot,
+  }
+  return await getDelegations(DELEGATIONS_ON_PROPOSAL_QUERY, variables)
+}
+
+const LATEST_DELEGATIONS_QUERY = `
```

Comment:
We can move all these queries to a file like... `src/entities/Snapshot/queries.ts` WDYT?